extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/HoodieCatalogRangerSparkExtensionSuite.scala

```diff
@@ -0,0 +1,125 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.plugin.spark.authz.ranger
+
+import org.apache.spark.SparkConf
+import org.scalatest.Outcome
+
+import org.apache.kyuubi.Utils
+import org.apache.kyuubi.plugin.spark.authz.AccessControlException
+import org.apache.kyuubi.plugin.spark.authz.RangerTestNamespace._
+import org.apache.kyuubi.plugin.spark.authz.RangerTestUsers._
+import org.apache.kyuubi.plugin.spark.authz.util.AuthZUtils._
+import org.apache.kyuubi.tags.HoodieTest
+
+/**
+ * Tests for RangerSparkExtensionSuite on Hoodie SQL.
+ */
+@HoodieTest
+class HoodieCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
+  override protected val catalogImpl: String = "in-memory"
+  override protected val sqlExtensions: String =
+    if (isSparkV31OrGreater) {
+      "org.apache.spark.sql.hudi.HoodieSparkSessionExtension"
+    } else {
+      ""
+    }
+
+  override protected val extraSparkConf: SparkConf =
+    new SparkConf()
+      .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+
+  val namespace1 = hoodieNamespace
+  val table1 = "table1_hoodie"
+  val table2 = "table2_hoodie"
+  val outputTable1 = "outputTable_hoodie"
+
+  override def withFixture(test: NoArgTest): Outcome = {
+    assume(isSparkV31OrGreater)
+    test()
+  }
+
+  override def beforeAll(): Unit = {
+    if (isSparkV31OrGreater) {
+      if (isSparkV32OrGreater) {
+        spark.conf.set(
+          s"spark.sql.catalog.$sparkCatalog",
+          "org.apache.spark.sql.hudi.catalog.HoodieCatalog")
+        spark.conf.set("hoodie.schema.on.read.enable", "true")
+        spark.conf.set(s"spark.sql.catalog.$sparkCatalog.type", "hadoop")
+        spark.conf.set(
+          s"spark.sql.catalog.$sparkCatalog.warehouse",
+          Utils.createTempDir("hoodie-hadoop").toString)
+      }
+      super.beforeAll()
+    }
+  }
+
+  override def afterAll(): Unit = {
+    if (isSparkV31OrGreater) {
+      super.afterAll()
+      spark.sessionState.catalog.reset()
+      spark.sessionState.conf.clear()
+    }
+  }
+
+  test("[KYUUBI #5284] Kyuubi authz support Hoodie Alter Table Command") {
+    doAs(admin, sql(s"CREATE DATABASE IF NOT EXISTS $namespace1"))
+    doAs(
+      admin,
+      sql(
+        s"""
+           |CREATE TABLE IF NOT EXISTS $namespace1.$table1(id int, name string, city string)
+           |USING hudi
+           |OPTIONS (
+           | type = 'cow',
+           | primaryKey = 'id',
+           | 'hoodie.datasource.hive_sync.enable' = 'false'
+           |)
+           |PARTITIONED BY(city)
+           |""".stripMargin))
+
+    val e1 = intercept[AccessControlException] {
+      doAs(someone, sql(s"ALTER TABLE $namespace1.$table1 ADD COLUMNS(age int)"))
+    }.getMessage
+    assert(e1.contains(s"does not have [alter] privilege" +
+      s" on [$namespace1/$table1/age]"))
```

Comment:
```suggestion
    interceptContains[AccessControlException] (
      doAs(someone, sql(s"ALTER TABLE $namespace1.$table1 ADD COLUMNS(age int)"))
    )(s"does not have [alter] privilege on [$namespace1/$table1/age]")
```
Use `AssertUtils.interceptContains` method instead for intercepting and string containing assertions.

Reply:
Done